hibernate-core/src/main/java/org/hibernate/persister/collection/AbstractCollectionPersister.java

```diff
@@ -309,7 +309,7 @@ public AbstractCollectionPersister(
 				final BasicValue discriminatorDescriptor = any.getDiscriminatorDescriptor();
 				final AnyType anyType = any.getType();
 				final MetaType metaType = (MetaType) anyType.getDiscriminatorType();
-				final Object discriminatorValue = metaType.getEntityNameToDiscriminatorValueMap().get( ownerPersister.getEntityName() );
+				Object discriminatorValue = metaType.resolveDiscriminatorValue( ownerPersister.getEntityName() );
```

Comment:
Please add the final back to match the rest of the code.
```suggestion
				final Object discriminatorValue = metaType.resolveDiscriminatorValue( ownerPersister.getEntityName() );
```

Reply:
Actually, as we discussed on Zulip, I think this shoud pass in the persister rather than "just" the name.  That would open up the possibility for the other things we discussed (short name, etc).